packages/terra-button/src/Button.jsx

```diff
@@ -133,6 +134,19 @@ class Button extends React.Component {
     }
   }
 
+  handleClick(event) {
```

Comment:
I assume this issue  of focus not shifting back to button in Safari and FireFox was due to button not being set as [previouslyFocusedElement](https://github.com/davidtheclark/focus-trap-react/blob/e67d26c5ff512228007bd6eb55aea9841eb7e09d/src/focus-trap-react.js#L10) in focus-trap and instead `main` element was being updated as previouslyFocusedElement. 

So Curious to know to keep button as active element and so that it gets updated as [previouslyFocusedElement](https://github.com/davidtheclark/focus-trap-react/blob/e67d26c5ff512228007bd6eb55aea9841eb7e09d/src/focus-trap-react.js#L10) when popup opens. do we need handleClick() event OR can we just set the focus to button on handleMouseDown() event like below.
```javascript
  handleMouseDown(event) {
    this.shouldShowFocus = false;
    setTimeout(() => { this.shouldShowFocus = true; });

    if (this.props.onMouseDown) {
      this.props.onMouseDown(event);
    }

    event.preventDefault();
    event.currentTarget.focus();
  }
```

Reply:
@supreethmr The issue is that in FF, Safari, and IE when you click on the button, it does not trigger the onFocus event like it does in Chrome. Since it didn't become focused after clicked, clicked outside would also not trigger the onBlur event, which causes the issue reported on #2688. Clicking and focus on a button varies by browser and OS. See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus. 

This change is to set the focus on the button when clicked. The mouseDown event is triggered on both left and right mouse click but the focus should only be set on the button being clicked. Hence, I need to do that in handleClick.

Reply:
Got it Thank you @benbcai  for the info.